subprojects/files/src/main/java/org/gradle/api/internal/file/collections/DefaultDirectoryWalker.java

```diff
@@ -114,7 +115,7 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
             return checkStopFlag();
         }
 
-        private FileVisitDetails getFileVisitDetails(Path file, BasicFileAttributes attrs, boolean isDirectory) {
+        private FileVisitDetails getFileVisitDetails(Path file, @Nullable BasicFileAttributes attrs, boolean isDirectory) {
```

Comment:
this _is_ actually nullable, unlike the snapshotter

Reply:
It's in contrast with: https://github.com/gradle/gradle/pull/9878/commits/1bfa9706044ca49a7dd67aa20e24aa962b9e3c9b